src/fsharp/CompileOps.fs

```diff
@@ -654,9 +654,12 @@ let OutputPhasedErrorR (os:System.Text.StringBuilder) (err:PhasedError) =
           match contextInfo with
           | ContextInfo.OmittedElseBranch -> os.Append(FSComp.SR.missingElseBranch(t2)) |> ignore
           | ContextInfo.ElseBranch -> os.Append(FSComp.SR.elseBranchHasWrongType(t1,t2)) |> ignore
-          | ContextInfo.TupleInRecordFields -> 
+          | ContextInfo.TupleInRecordFields ->
+                os.Append(ErrorFromAddingTypeEquation1E().Format t2 t1 tpcs) |> ignore
+                os.Append(System.Environment.NewLine + FSComp.SR.commaInsteadOfSemicolonInRecord()) |> ignore
+          | _ when t2 = "bool" && t1.EndsWith " ref" ->
```

Comment:
A bit of a hacky way to get this condition, but I suppose it's good enough :)  It's hard to imagine it no working.